src/lib/logical_query_plan/compare_lqps.cpp

```diff
@@ -0,0 +1,291 @@
+#include "compare_lqps.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/create_view_node.hpp"
+#include "logical_query_plan/delete_node.hpp"
+#include "logical_query_plan/drop_view_node.hpp"
+#include "logical_query_plan/dummy_table_node.hpp"
+#include "logical_query_plan/insert_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/show_columns_node.hpp"
+#include "logical_query_plan/show_tables_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "logical_query_plan/update_node.hpp"
+#include "logical_query_plan/validate_node.hpp"
+
+namespace opossum {
+
+SemanticLQPCompare::SemanticLQPCompare(const std::shared_ptr<const AbstractLQPNode>& lhs, const std::shared_ptr<const AbstractLQPNode>& rhs): _lhs(lhs), _rhs(rhs) {
```

Comment:
b) I'm not sure about the "semantic" name. To me, this does not look like [semantic equivalence](https://en.wikipedia.org/wiki/Semantic_equivalence) but like something stricter. Isn't this more an identity vs equality question?

Reply:
Don't like "semantic" here either, even though I don't think its totally off. "Compare" also feels wrong, since we're not ordering LQPs according to some `<` property.

*Probably* this is an equality check of a complex structure (the node, its subnodes AND the also rather involved comparison of two LQPColumnReferences from different LQPs).

So, `class LQPEquals` or `bool lqp_equals()`? I'd like to keep the tree traversal out of the LQP code and maybe just put the node equality checks there, e.g. `AggregateNode::equals(AnotherAggregateNode)`

Reply:
Why would you want to have an external method for that? The tree traversal should be five lines.

How do you feel about `AbstractLQPNode::deep_equals` and `virtual AbstractLQPNode::shallow_equals`?

Reply:
not too well, but I get that you like class member methods and will do it this way for now